clients/IntermediaryClient.ts

```diff
@@ -174,11 +174,17 @@ export class IntermediaryClient extends StateChannelWallet {
       // todo: account for expected gas consumption? ( out of scope for hackathon )
       throw new Error("Transfer amount exceeds owner balance");
     }
-
     const ownerSig = userOp.signature;
     const { signature: intermediarySig } = await this.signUserOperation(userOp);
     userOp.signature = ethers.concat([ownerSig, intermediarySig]);
 
+    // TODO: We expect this validate call to pass or revert
```

Comment:
In the contract tests, a check like this either returns 0 or throws an error with the revert reason. However in the client we get the error:
```
errors.ts:694 Uncaught (in promise) Error: could not decode result data (value="0x", info={ "method": "validateUserOp", "signature": "validateUserOp((address,uint256,bytes,bytes,uint256,uint256,uint256,uint256,uint256,bytes,bytes),bytes32,uint256)" }, code=BAD_DATA, version=6.8.0)
    at makeError (errors.ts:694:21)
    at assert (errors.ts:715:25)
    at _Interface.decodeFunctionResult (interface.ts:916:9)
    at staticCallResult (contract.ts:346:35)
    at async AsyncFunction.staticCall (contract.ts:303:24)
```

Reply:
In the tests it looks like we are setting the `UserOperation` values to the following:
```
    const userOp: UserOperationStruct = {
      sender: await nitroSCW.getAddress(),
      nonce: 0,
      initCode: "0x",
      callData,
      callGasLimit: 40_000,
      verificationGasLimit: 150000,
      preVerificationGas: 21000,
      maxFeePerGas: 40_000,
      maxPriorityFeePerGas: 40_000,
      paymasterAndData: hre.ethers.ZeroHash,
      signature: hre.ethers.ZeroHash,
    };
```
The default values set in `UserOp.ts` by the `fillUserOpDefaults` are:
```
export const DefaultsForUserOp: UserOperationStruct = {
  sender: ZeroAddress,
  nonce: 0,
  initCode: "0x",
  callData: "0x",
  callGasLimit: 0,
  verificationGasLimit: 150000, // default verification gas. will add create2 cost (3200+200*length) if initCode exists
  preVerificationGas: 21000, // should also cover calldata cost.
  maxFeePerGas: 0,
  maxPriorityFeePerGas: 1e9,
  paymasterAndData: "0x",
  signature: "0x",
};
```
In `OwnerClient.payL1` the only defaults we override are `sender` and `calldata`:
```
    const partialUserOp: Partial<UserOperationStruct> = {
      sender: this.signer.address,
      callData,
    };
    const userOp = fillUserOpDefaults(partialUserOp);
```
Perhaps that discrepancy is causing an issue?

Reply:
It looks like the underlying issue was a problem with the deployment. However I have updated:
- The `UserOperation` to use the scwAddress as the sender
- The `UserOperation` to use the same gas values from the test.